app/gui/view/graph-editor/src/builtin/visualization/native/text_grid.rs

```diff
@@ -0,0 +1,432 @@
+pub struct Entry {
+    dummy_root: display::object::Instance,
+    text:       Rc<DomSymbol>,
+    frp:        Rc<EntryFrp<Self>>,
+}
+
+impl Entry {
+    fn set_model(&self, model: &entry::Model) {
+        self.text.set_inner_text(&model.text);
+    }
+
+    fn set_params(&self, params: &entry::Params) {
+        if let Some(parent) = &params.parent {
+            parent.append_or_warn(self.text.dom());
+        }
+        self.text.set_style_or_warn("font-family", params.font_name.clone());
+        self.text.set_style_or_warn("font-size", format!("{}px", params.font_size as u32));
+    }
+
+    fn set_position_and_size(&self, pos: &Vector2, size: &Vector2) {
+        self.text.set_position_xy(*pos);
+
+        self.text.set_style_or_warn("left", format!("{}px", pos.x - size.x / 2.0));
+        self.text.set_style_or_warn("top", format!("{}px", -pos.y - size.y / 2.0));
+
+        self.text.set_style_or_warn("width", format!("{}px", size.x as u32));
+        self.text.set_style_or_warn("height", format!("{}px", size.y as u32));
+    }
+}
+
+impl display::Object for Entry {
+    fn display_object(&self) -> &display::object::Instance {
+        &self.dummy_root
+    }
+}
+
+impl grid_view::Entry for Entry {
+    type Model = entry::Model;
+    type Params = entry::Params;
+
+    fn new(app: &Application, _text_layer: Option<&display::scene::Layer>) -> Self {
+        let scene = &app.display.default_scene;
+        let root = display::object::Instance::new();
+        let text_div = web::document.create_div_or_panic();
+        let text = DomSymbol::new(&text_div);
+        scene.dom.layers.back.manage(&text);
+        text.set_style_or_warn("white-space", "nowrap");
+        text.set_style_or_warn("pointer-events", "auto");
+        text.set_style_or_warn("white-space", "pre");
+
+        let new_entry = Self { text: Rc::new(text), frp: default(), dummy_root: root };
+
+        let input = &new_entry.frp.private().input;
+        let network = new_entry.frp.network();
+        enso_frp::extend! { network
+            init <- source_();
+            eval input.set_model((model) new_entry.set_model(model));
+            eval input.set_params((params) new_entry.set_params(params));
+
+            pos_size <- all(&input.set_position, &input.set_size);
+            eval pos_size (((pos, size)) new_entry.set_position_and_size(pos, size));
+        }
+        init.emit(());
+        new_entry
+    }
+
+    fn frp(&self) -> &EntryFrp<Self> {
+        &self.frp
+    }
+}
+
+
+
+// =============
+// === Model ===
+// =============
+
+#[derive(Debug)]
+#[allow(missing_docs)]
+pub struct Model<T> {
+    app:                   Application,
+    size:                  Rc<Cell<Vector2>>,
+    root:                  display::object::Instance,
+    text_grid:             GridView<Entry>,
+    dom_entry_root:        DomSymbol,
+    clipping_div:          DomSymbol,
+    scroll_bar_horizontal: Scrollbar,
+    scroll_bar_vertical:   Scrollbar,
```

Comment:
As we had discussed, this is not using `scrollable::GridView` to avoid adding a `ScrollArea` and the associated masking layer. I've added a comment to that effect.